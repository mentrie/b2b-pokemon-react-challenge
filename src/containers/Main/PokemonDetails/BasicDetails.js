import React from 'react';
import PropTypes from 'prop-types';
import backArrowIcon from '../../../static/images/back-arrow.svg';
import PokemonItem from '../Pokemons/PokemonItem';
import { formatName } from '../../../helpers/utils';

const BasicDetails = ({ details, backToHome }) => {
  return (
    <div className="pokemon__details">
      <div>
        <img src={backArrowIcon} onClick={backToHome} alt="back arrow" />
        <span className="pokemon__name">{formatName(details.name)}</span>
      </div>
      <div className="pokemon__basic__details">
        <PokemonItem pokemon={details.forms[0]} />
        <div className="pokemon__details__items">
          <div className="basic__item">
            <span>Name:</span>
            <span>{details.name}</span>
          </div>
          <div className="basic__item">
            <span>Weight:</span>
            <span>{details.weight}</span>
          </div>
          <div className="basic__item">
            <span>{`Type${details.types.length > 1 ? 's' : ''}`}:</span>
            <span>{details.types.map(({type}) => type.name).join(', ')}</span>
          </div>
          <div className="basic__item">
            <span>{`Abilit${details.abilities.length > 1 ? 'ies' : 'y'}`}:</span>
            <span>{details.abilities.map(({ability}) => ability.name).join(', ')}</span>
          </div>
          <div className="basic__item">
            <span>Evolution Info:</span>
            <span>{details.base_experience}</span>
          </div>
        </div>
      </div>
    </div>
  );
};

BasicDetails.propTypes = {
  details: PropTypes.shape({
    name: PropTypes.string,
    types: PropTypes.array,
    abilities: PropTypes.array,
    base_experience: PropTypes.number
  }),
  backToHome: PropTypes.func.isRequired
};

export default BasicDetails;
