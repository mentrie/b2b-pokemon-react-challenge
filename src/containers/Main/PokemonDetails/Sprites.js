import React from 'react';

const Sprites = ({ sprites }) => {
  const spritesList = Object.keys(sprites).filter(key => sprites[key]);

  if (spritesList.length > 0) {
    const spritesToRender = spritesList.map((sprite, index) => (
      <div key={index} className="sprite__item">
        <div className="sprite__background__cover" style={{backgroundImage: `url(${sprites[sprite]})`}} />
        <img className="sprite__picture" alt={`sprint-${index}`} src={sprites[sprite]} />
      </div>
    ));

    return (
      <div className="sprites__wrapper">
        <h3 className="sprites__title">Sprites</h3>
        <div className="sprites__list">
          {spritesToRender}
        </div>
      </div>
    );
  }
};

export default Sprites;
