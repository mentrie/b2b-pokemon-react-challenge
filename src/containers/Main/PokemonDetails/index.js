import React, { Component } from 'react';
import { apiRequest } from '../../../api';

class PokemonDetails extends Component {
  state = {
    details: null
  }

  async componentWillMount() {
    if (this.state.details == null) {
      const { match } = this.props;
    //   try {
    //     const result = await apiRequest('/')
    //   }
    }
  }
  render() {
    return (
      <div>
        <p onClick={this.props.backToHome}> Back </p>
        Details
      </div>
    );
  }
};

export default PokemonDetails;
