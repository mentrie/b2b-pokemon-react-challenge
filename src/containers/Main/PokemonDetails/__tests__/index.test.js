import React from 'react';
import PokemonDetails from '../index';
import Router from 'react-router-dom/Router';
import createHistory from 'history/createBrowserHistory';

test('render component properly', () => {
  const component = shallow(<PokemonDetails />);

  expect(component).toMatchSnapshot();
});

test('render subcomponents', () => {
  const history = createHistory();
  const component = mount(<Router history={history}><PokemonDetails /></Router>);

  expect(component).toMatchSnapshot();
});
