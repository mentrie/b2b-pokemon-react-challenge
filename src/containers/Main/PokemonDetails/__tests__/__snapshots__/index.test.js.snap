// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component properly 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`render subcomponents 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <withRouter(PokemonDetails)>
    <Route>
      <PokemonDetails
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
      >
        <ReactPlaceholder
          color="#CDCDCD"
          delay={0}
          ready={false}
          rows={7}
          showLoadingAnimation={true}
          type="media"
        >
          <MediaBlock
            className="show-loading-animation"
            color="#CDCDCD"
            delay={0}
            rows={7}
          >
            <div
              className="media-block show-loading-animation"
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <RoundShape
                color="#CDCDCD"
                style={
                  Object {
                    "marginRight": 10,
                    "minHeight": 55,
                    "minWidth": 55,
                    "width": 55,
                  }
                }
              >
                <div
                  className="round-shape"
                  style={
                    Object {
                      "backgroundColor": "#CDCDCD",
                      "borderRadius": "500rem",
                      "height": "100%",
                      "marginRight": 10,
                      "minHeight": 55,
                      "minWidth": 55,
                      "width": 55,
                    }
                  }
                />
              </RoundShape>
              <TextBlock
                color="#CDCDCD"
                rows={7}
                widths={
                  Array [
                    97,
                    100,
                    94,
                    90,
                    98,
                    95,
                    98,
                    40,
                  ]
                }
              >
                <div
                  className="text-block"
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                >
                  <TextRow
                    key="0"
                    lineSpacing={0}
                    style={
                      Object {
                        "backgroundColor": "#CDCDCD",
                        "maxHeight": "7.6923076923076925%",
                        "width": "97%",
                      }
                    }
                  >
                    <div
                      className="text-row"
                      style={
                        Object {
                          "backgroundColor": "#CDCDCD",
                          "height": "1em",
                          "marginTop": 0,
                          "maxHeight": "7.6923076923076925%",
                          "width": "97%",
                        }
                      }
                    />
                  </TextRow>
                  <TextRow
                    key="1"
                    lineSpacing="0.7em"
                    style={
                      Object {
                        "backgroundColor": "#CDCDCD",
                        "maxHeight": "7.6923076923076925%",
                        "width": "100%",
                      }
                    }
                  >
                    <div
                      className="text-row"
                      style={
                        Object {
                          "backgroundColor": "#CDCDCD",
                          "height": "1em",
                          "marginTop": "0.7em",
                          "maxHeight": "7.6923076923076925%",
                          "width": "100%",
                        }
                      }
                    />
                  </TextRow>
                  <TextRow
                    key="2"
                    lineSpacing="0.7em"
                    style={
                      Object {
                        "backgroundColor": "#CDCDCD",
                        "maxHeight": "7.6923076923076925%",
                        "width": "94%",
                      }
                    }
                  >
                    <div
                      className="text-row"
                      style={
                        Object {
                          "backgroundColor": "#CDCDCD",
                          "height": "1em",
                          "marginTop": "0.7em",
                          "maxHeight": "7.6923076923076925%",
                          "width": "94%",
                        }
                      }
                    />
                  </TextRow>
                  <TextRow
                    key="3"
                    lineSpacing="0.7em"
                    style={
                      Object {
                        "backgroundColor": "#CDCDCD",
                        "maxHeight": "7.6923076923076925%",
                        "width": "90%",
                      }
                    }
                  >
                    <div
                      className="text-row"
                      style={
                        Object {
                          "backgroundColor": "#CDCDCD",
                          "height": "1em",
                          "marginTop": "0.7em",
                          "maxHeight": "7.6923076923076925%",
                          "width": "90%",
                        }
                      }
                    />
                  </TextRow>
                  <TextRow
                    key="4"
                    lineSpacing="0.7em"
                    style={
                      Object {
                        "backgroundColor": "#CDCDCD",
                        "maxHeight": "7.6923076923076925%",
                        "width": "98%",
                      }
                    }
                  >
                    <div
                      className="text-row"
                      style={
                        Object {
                          "backgroundColor": "#CDCDCD",
                          "height": "1em",
                          "marginTop": "0.7em",
                          "maxHeight": "7.6923076923076925%",
                          "width": "98%",
                        }
                      }
                    />
                  </TextRow>
                  <TextRow
                    key="5"
                    lineSpacing="0.7em"
                    style={
                      Object {
                        "backgroundColor": "#CDCDCD",
                        "maxHeight": "7.6923076923076925%",
                        "width": "95%",
                      }
                    }
                  >
                    <div
                      className="text-row"
                      style={
                        Object {
                          "backgroundColor": "#CDCDCD",
                          "height": "1em",
                          "marginTop": "0.7em",
                          "maxHeight": "7.6923076923076925%",
                          "width": "95%",
                        }
                      }
                    />
                  </TextRow>
                  <TextRow
                    key="6"
                    lineSpacing="0.7em"
                    style={
                      Object {
                        "backgroundColor": "#CDCDCD",
                        "maxHeight": "7.6923076923076925%",
                        "width": "98%",
                      }
                    }
                  >
                    <div
                      className="text-row"
                      style={
                        Object {
                          "backgroundColor": "#CDCDCD",
                          "height": "1em",
                          "marginTop": "0.7em",
                          "maxHeight": "7.6923076923076925%",
                          "width": "98%",
                        }
                      }
                    />
                  </TextRow>
                </div>
              </TextBlock>
            </div>
          </MediaBlock>
        </ReactPlaceholder>
      </PokemonDetails>
    </Route>
  </withRouter(PokemonDetails)>
</Router>
`;
